pytorch_lightning/trainer/trainer.py

```diff
@@ -713,17 +713,38 @@ def tng_tqdm_dic(self):
     # -----------------------------
     # MODEL TRAINING
     # -----------------------------
-    def fit(self, model):
+    def fit(self, model, train_dataloader=None, val_dataloader=None, test_dataloader=None):
         r"""
         Runs the full optimization routine.
 
-        Example::
+        Args:
+            model (LightningModule): Model to fit.
+            Example::
 
-            trainer = Trainer()
-            model = LightningModule()
+                trainer = Trainer()
```

Comment:
write two examples:

Example 1: no dataloaders passed, then say that it will use the train_dataloader, etc... calls.

Example 2: with the .fit(...)

In addition, elaborate as to why we might want to do one  vs the other